src/lib/Day2/IntcodeRunner.cs

```diff
@@ -44,36 +55,142 @@ public int[] Execute()
         public int[] Execute(int noun, int verb)
         {
             var output = (int[])this.instructions.Clone();
+            var outputIndex = -1;
             output[1] = noun;
             output[2] = verb;
 
-            int index = 0;
-            var opcode = (Opcode)output[index];
+            int index = 0, numParams = 0;            
+            var (opcode, operModes) = ParseOpcode(output[index]);
+
             while (opcode != Opcode.End)
             {
-                this.Execute(opcode, output[index + 1], output[index + 2], output[index + 3], output);
-                index += 4;
-                opcode = (Opcode)output[index];
+                switch (opcode)
+                {
+                    case Opcode.Add:
+                    case Opcode.Multiply:
+                    case Opcode.LessThan:
+                    case Opcode.Equals:
+                        numParams = 2;
+                        outputIndex = index + numParams + 1;
+                        break;
+                    case Opcode.JumpIfTrue:
+                    case Opcode.JumpIfFalse:
+                        numParams = 2;
+                        outputIndex = index + numParams;
+                        break;
+                    case Opcode.Read:
+                    case Opcode.Print:
+                        numParams = 1;
+                        outputIndex = index + numParams;
+                        break;
+                }
+
```

Comment:
Use outputIndexOffset in switch statement

    case Opcode.Equals:
        outputIndexOffset = 1;

Consolidate outputIndex = index + numParams here    

```suggestion
outputIndex += index + numParams + outputIndexOffset;
```